providers/mthca/verbs.c

```diff
@@ -709,6 +709,13 @@ int mthca_destroy_qp(struct ibv_qp *qp)
 struct ibv_ah *mthca_create_ah(struct ibv_pd *pd, struct ibv_ah_attr *attr)
 {
 	struct mthca_ah *ah;
+	struct ibv_port_attr port_attr;
+
+	if (mthca_query_port(pd->context, attr->port_num, &port_attr))
+		return NULL;
+
+	if (port_attr.flags & IBV_QPF_GRH_REQUIRED && !attr->is_global)
+		return NULL;
```

Comment:
What is the point of this? The mthca driver never sets RDMA_CORE_CAP_IB_GRH_REQUIRED and isn't roce..

Reply:
The idea was to be consistent for all drivers and have it ready from day one for a case that some kernel support will be added.

Reply:
mthca is never going to be changed, and doing it in this hacky way creates a big performance regression for it, prefer to see it dropped I think..

Reply:
OK, will drop the mthca part from this patch.